app/views/advanced_messenger/_html_head.html.erb

```diff
@@ -58,20 +58,18 @@
 
         function renderExpandedCollapsed(entityId, endpoint, currentNode, expanded, isInitialRender) {
             renderExpandCollapseButton(expanded, currentNode, entityId, endpoint);
+            if (<%=AdvancedMessengerSettings.get_setting(:disable_fix_for_scroll_to_anchor) != '1'%> && isInitialRender) {
+                return;
+            }
 
-            var showOrHideContent = expanded ? "show" : "hide";
-            var showOrHidePreview = expanded ? "hide" : "show";
-
+            var showOrHide = (node, isVisible) => isVisible ? node.removeClass("hidden") : node.addClass("hidden") 
             // ======= Show or hide content ======
-            currentNode.find('.message-content')[showOrHideContent]();
-
-            currentNode.find('.details')[showOrHideContent]();
-            // setTimeout is a workarround for:
-            // At page load, for collapsed notes with images, even if we hide them they become visible again
-            setTimeout(() => currentNode.find('.thumbnails')[showOrHideContent](), isInitialRender && !expanded ? 100 : 0);
-            
+            showOrHide(currentNode.find('.message-content'), expanded);
```

Comment:
Initialy, we were hiding/showing the note by calling the jquery show()/hide() functions. 
Now on the server side we hide/show them by adding/removing the "hidden" css class. I changed here also to be uniform with the server side code.
in addition, I removed the setTimeout() workarround because it was needed only in case of initial page load and this function was not called anymore in this case, the note being expanded on the server side. 
But indeed I should place back the setTimeout() know that we have a settings for disabling the server side collapse

Reply:
I testes again, and with the new modifications, the setTimeout() hack is not needed anymore because our class: .hidden is more important that the style display:block, added somehow delayed by the redmine application